### PR TITLE
Test each gateway

### PR DESCRIPTION
**Pull request by scothis**
builds on #13

### CODE REVIEW

**Comment by scothis** [comment]
Eventually, I'd like to get to the point where we don't spin up k8s or riff to test the client. Instead consuming the liiklus docker image directly. Something along the lines of:

```
docker run --rm -it -p 6565:6565/tcp -p 8080:8080/tcp -e storage_positions_type=MEMORY -e storage_records_type=MEMORY bsideup/liiklus:0.9.1
```

For now, we're hunting enough bugs that are exposed by different brokers that it's worth full integration tests.

File: client.go [hunk 1]
```diff
@@ -130,66 +130,72 @@ func chopContentType(contentType string) string {
...
-				Offset:               record.Offset,
-			}
-			_, err = lc.client.Ack(subContext, &ackRequest)
+			receiveClient, err := lc.client.Receive(subContext, &receiveRequest)
 			if err != nil {
 				e(cancel, err)
 				return
 			}
+
+			go func() {
```
Comment (sbawaska) [inline client.go:160]: why do we have nested go routines here? Was this the issue with subscribe not working?

Comment (scothis) [reply client.go:160]: we need to receive messages from each partition, not just the first partition. This is the crux of why the in-memory gateway appeared to not work.

**Comment by sbawaska** [comment]
LGTM!

**Review by sbawaska — APPROVED** [review]

**Pull request merged**